commerce-product-type-virtual-order-service/src/main/java/com/liferay/commerce/product/type/virtual/order/service/impl/CommerceVirtualOrderItemLocalServiceImpl.java

```diff
@@ -248,8 +248,8 @@ public File getFile(long commerceVirtualOrderItemId) throws Exception {
 			commerceOrderItem.getNameCurrentValue() + StringPool.PERIOD +
 				extension);
 
-		if (file.exists()) {
-			file.delete();
+		if (file.exists() && !file.delete()) {
+			throw new Exception();
```

Comment:
It'd be nice to throw a more specific exception here... maybe even just `IOException`?